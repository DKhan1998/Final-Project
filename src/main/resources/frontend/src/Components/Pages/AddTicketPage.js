import React from "react";
import AddTicketForm from '../tickets/AddTicketForm'
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faPlusSquare} from "@fortawesome/free-solid-svg-icons";

const AddTicketPage = () => {
    return (
        <div>
            <h1 id="pageheader"> New Ticket</h1>
            <AddTicketForm/>

        </div>
    )

}
export default AddTicketPage;