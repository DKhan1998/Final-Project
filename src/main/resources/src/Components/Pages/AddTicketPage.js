import React from "react";
import AddTicket from '../tickets/AddTicket'

const AddTicketPage = () =>  {
    return(
        <div>
            <h1 id="pageheader"> New Ticket</h1>
            <AddTicket/>

        </div>
    )

}
export default AddTicketPage;