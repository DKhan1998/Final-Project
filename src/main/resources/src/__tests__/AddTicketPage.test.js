import React from "react";
import AddTicketPage from "../Components/Pages/AddTicketPage"
import render from "react-test-renderer";

//snapshot test
test("AddTicketPage Snapshot Test", () => {
    const tree = render.create(<AddTicketPage/>).toJSON();
    expect(tree).toMatchSnapshot();
})