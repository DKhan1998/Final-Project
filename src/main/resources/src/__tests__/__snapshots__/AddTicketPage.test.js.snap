// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTicketPage Snapshot Test 1`] = `
<div>
  <h1
    id="pageheader"
  >
     New Ticket
  </h1>
  <div>
    <div
      style={
        Object {
          "display": "none",
        }
      }
    >
      <div
        style={null}
      />
    </div>
    <form
      className=""
      id="ticketForm"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="form-row"
      >
        <div
          className="form-group col"
        >
          <label
            className="form-label"
            htmlFor="formCohort"
          >
            Cohort
          </label>
          <select
            autoComplete="off"
            className="form-control"
            defaultValue="Choose..."
            id="formCohort"
            name="cohort"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option>
              Choose...
            </option>
            <option
              value="Azure-Devops"
            >
              Azure Devops - (20OctDevOps1)
            </option>
            <option
              value="Software-BAE"
            >
              Software BAE - (20OCTBAE2)
            </option>
            <option
              value="SDET"
            >
              SDET - (20OCTSDET3)
            </option>
            <option
              value="Cloud-Native-Engineer"
            >
              Cloud Native Engineer - (20SeptCNE)
            </option>
            <option
              value="Scala"
            >
              Scala - (20NovSoftware1)
            </option>
          </select>
        </div>
        <div
          className="form-group col"
        >
          <label
            className="form-label"
            htmlFor="formAuthor"
          >
            Author
          </label>
          <input
            autoComplete="off"
            className="form-control"
            id="formAuthor"
            name="author"
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formTitle"
        >
          Issue Title
        </label>
        <input
          autoComplete="off"
          className="form-control"
          id="formTitle"
          name="title"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formDescription"
        >
          Issue Description
        </label>
        <textarea
          autoComplete="off"
          className="form-control"
          id="formDescription"
          name="description"
          onChange={[Function]}
          required={true}
          rows="4"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formLinks"
        >
          Links
        </label>
        <input
          autoComplete="off"
          className="form-control"
          id="formLinks"
          name="links"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <button
        className="btn btn-success"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-save fa-w-14 "
          data-icon="save"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M433.941 129.941l-83.882-83.882A48 48 0 0 0 316.118 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V163.882a48 48 0 0 0-14.059-33.941zM224 416c-35.346 0-64-28.654-64-64 0-35.346 28.654-64 64-64s64 28.654 64 64c0 35.346-28.654 64-64 64zm96-304.52V212c0 6.627-5.373 12-12 12H76c-6.627 0-12-5.373-12-12V108c0-6.627 5.373-12 12-12h228.52c3.183 0 6.235 1.264 8.485 3.515l3.48 3.48A11.996 11.996 0 0 1 320 111.48z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        SAVE
      </button>
        
      <button
        className="btn btn-info"
        disabled={false}
        type="reset"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-undo fa-w-16 "
          data-icon="undo"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M212.333 224.333H12c-6.627 0-12-5.373-12-12V12C0 5.373 5.373 0 12 0h48c6.627 0 12 5.373 12 12v78.112C117.773 39.279 184.26 7.47 258.175 8.007c136.906.994 246.448 111.623 246.157 248.532C504.041 393.258 393.12 504 256.333 504c-64.089 0-122.496-24.313-166.51-64.215-5.099-4.622-5.334-12.554-.467-17.42l33.967-33.967c4.474-4.474 11.662-4.717 16.401-.525C170.76 415.336 211.58 432 256.333 432c97.268 0 176-78.716 176-176 0-97.267-78.716-176-176-176-58.496 0-110.28 28.476-142.274 72.333h98.274c6.627 0 12 5.373 12 12v48c0 6.627-5.373 12-12 12z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         Reset Form
      </button>
    </form>
  </div>
</div>
`;

exports[`it matches the snapshot 1`] = `
<div>
  <h1
    id="pageheader"
  >
     New Ticket
  </h1>
  <div>
    <div
      style={
        Object {
          "display": "none",
        }
      }
    >
      <div
        style={null}
      />
    </div>
    <form
      className=""
      id="ticketForm"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="form-row"
      >
        <div
          className="form-group col"
        >
          <label
            className="form-label"
            htmlFor="formCohort"
          >
            Cohort
          </label>
          <select
            autoComplete="off"
            className="form-control"
            defaultValue="Choose..."
            id="formCohort"
            name="cohort"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option>
              Choose...
            </option>
            <option
              value="Azure-Devops"
            >
              Azure Devops - (20OctDevOps1)
            </option>
            <option
              value="Software-BAE"
            >
              Software BAE - (20OCTBAE2)
            </option>
            <option
              value="SDET"
            >
              SDET - (20OCTSDET3)
            </option>
            <option
              value="Cloud-Native-Engineer"
            >
              Cloud Native Engineer - (20SeptCNE)
            </option>
            <option
              value="Scala"
            >
              Scala - (20NovSoftware1)
            </option>
          </select>
        </div>
        <div
          className="form-group col"
        >
          <label
            className="form-label"
            htmlFor="formAuthor"
          >
            Author
          </label>
          <input
            autoComplete="off"
            className="form-control"
            id="formAuthor"
            name="author"
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formTitle"
        >
          Issue Title
        </label>
        <input
          autoComplete="off"
          className="form-control"
          id="formTitle"
          name="title"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formDescription"
        >
          Issue Description
        </label>
        <textarea
          autoComplete="off"
          className="form-control"
          id="formDescription"
          name="description"
          onChange={[Function]}
          required={true}
          rows="4"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formLinks"
        >
          Links
        </label>
        <input
          autoComplete="off"
          className="form-control"
          id="formLinks"
          name="links"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <button
        className="btn btn-success"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-save fa-w-14 "
          data-icon="save"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M433.941 129.941l-83.882-83.882A48 48 0 0 0 316.118 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V163.882a48 48 0 0 0-14.059-33.941zM224 416c-35.346 0-64-28.654-64-64 0-35.346 28.654-64 64-64s64 28.654 64 64c0 35.346-28.654 64-64 64zm96-304.52V212c0 6.627-5.373 12-12 12H76c-6.627 0-12-5.373-12-12V108c0-6.627 5.373-12 12-12h228.52c3.183 0 6.235 1.264 8.485 3.515l3.48 3.48A11.996 11.996 0 0 1 320 111.48z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        SAVE
      </button>
        
      <button
        className="btn btn-info"
        disabled={false}
        type="reset"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-undo fa-w-16 "
          data-icon="undo"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M212.333 224.333H12c-6.627 0-12-5.373-12-12V12C0 5.373 5.373 0 12 0h48c6.627 0 12 5.373 12 12v78.112C117.773 39.279 184.26 7.47 258.175 8.007c136.906.994 246.448 111.623 246.157 248.532C504.041 393.258 393.12 504 256.333 504c-64.089 0-122.496-24.313-166.51-64.215-5.099-4.622-5.334-12.554-.467-17.42l33.967-33.967c4.474-4.474 11.662-4.717 16.401-.525C170.76 415.336 211.58 432 256.333 432c97.268 0 176-78.716 176-176 0-97.267-78.716-176-176-176-58.496 0-110.28 28.476-142.274 72.333h98.274c6.627 0 12 5.373 12 12v48c0 6.627-5.373 12-12 12z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         Reset Form
      </button>
    </form>
  </div>
</div>
`;
